draft-barnes-acme.md

```diff
@@ -516,18 +544,211 @@ Link: <https://example.com/acme/terms>;rel="terms-of-service"
 
 If the client wishes to update this information in the future, it sends a POST request with updated information to the registration URI.  The server MUST ignore any updates to the "key", "authorizations, or "certificates" fields, and MUST verify that the request is signed with the private key corresponding to the "key" field of the request before updating the registration.
 
-Servers SHOULD NOT respond to GET requests for registration resources as these requests are not authenticated.  If a client wishes to query the server for information about its account (e.g., to examine the "contact" or "certificates" fields), then it SHOULD do so by sending a POST request with an empty update.  That is, it should send a JWS whose payload is the empty JSON dictionary ("{}").
+Servers SHOULD NOT respond to GET requests for registration resources as these requests are not authenticated.  If a client wishes to query the server for information about its account (e.g., to examine the "contact" or "certificates" fields), then it SHOULD do so by sending a POST request with an empty update.  That is, it should send a JWS whose payload is trivial ({"resource":"reg"}).
+
+### Recovery Keys
+
+If the client wishes to establish a secret key with the server that it can use to recover this account later (a "recovery key"), then it must perform a simple key agreement protocol as part of the new-registration transaction.  The client and server perform an ECDH exchange through the new-registration transaction, and the result is the recovery key.
+
+To request a recovery key, the client includes a "recoveryKey" field in its new-registration request.  The value of this field is a JSON object.
+
+client (required, JWK):
+: The client's ECDH public key
+
+server (optional, JWK):
+: The server's ECDH public key
+
+length (optional, number):
+: The length, in bits, of the derived secret.  The value MUST be a multiple of 8.
+
+confirm (optional, string):
+: The confirmation value, base64-encoded
```

Comment:
Ah. I read this section as being a value that you could use to determine whether you had correctly derived the same shared secret, to confirm that the recovery will work in the future.